File: jssdk-java-server/src/main/java/com/qiyuesuo/proxy/http/StreamFile.java
package com.qiyuesuo.proxy.http;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public class StreamFile implements FileItem {

    private String fileName;
    private InputStream stream;
    private String mimeType;
    private int printCount;

    private static final String DEFAULT_FILE_NAME = "streamFile";

    public StreamFile(String fileName, InputStream stream, String mimeType) {
        this.fileName = fileName;
        this.stream = stream;
        this.mimeType = mimeType;
    }

    public StreamFile(String fileName, InputStream stream) {
        this.fileName = fileName;
        this.stream = stream;
        this.mimeType = MIME_TYPE_DEFAULT;
    }

    public StreamFile(InputStream stream) {
        this.fileName = DEFAULT_FILE_NAME;
        this.stream = stream;
        this.mimeType = MIME_TYPE_DEFAULT;
    }

    public InputStream getStream() {
        return stream;
    }

    @Override
    public boolean isValid() {
        return this.stream != null && this.fileName != null;
    }

    @Override
    public String getFileName() {
        return this.fileName;
    }

    @Override
    public String getMimeType() {
        if (this.mimeType == null) {
            return MIME_TYPE_DEFAULT;
        } else {
            return this.mimeType;
        }
    }

    @Override
    public long getFileLength() {
        return 0L;
    }

    @Override
    public void write(OutputStream output) throws IOException {
        try {
            byte[] buffer = new byte[READ_BUFFER_SIZE];
            int n = 0;
            while (-1 != (n = stream.read(buffer))) {
                output.write(buffer, 0, n);
            }
        } finally {
            if (stream != null) {
                stream.close();
            }
        }
    }

    public int getPrintCount() {
        return printCount;
    }

    public void setPrintCount(int printCount) {
        this.printCount = printCount;
    }

}

File: jssdk-csharp-server/proxy-csharp/Tools/StreamFile.cs
using System;
using System.IO;

namespace proxy_csharp.Tools
{
    public class StreamFile : IFileItem
    {
        /** 默认媒体类型 **/
        public static string MIME_TYPE_DEFAULT = "application/octet-stream";
        /** 默认流式读取缓冲区大小 **/
        public static int READ_BUFFER_SIZE = 1024 * 4;

        private string fileName;
        private Stream stream;
        private string mimeType;

        private static string DEFAULT_FILE_NAME = "streamFile";

        public StreamFile(string fileName, Stream stream, string mimeType)
        {
            this.fileName = fileName;
            this.stream = stream;
            this.mimeType = mimeType;
        }

        public StreamFile(string fileName, Stream stream)
        {
            this.fileName = fileName;
            this.stream = stream;
            this.mimeType = MIME_TYPE_DEFAULT;
        }

        public StreamFile(Stream stream)
        {
            this.fileName = DEFAULT_FILE_NAME;
            this.stream = stream;
            this.mimeType = MIME_TYPE_DEFAULT;
        }

        public long GetFileLength()
        {
            return 0L;
        }

        public string GetFileName()
        {
            return this.fileName;
        }

        public string GetMimeType()
        {
            if (this.mimeType == null)
            {
                return MIME_TYPE_DEFAULT;
            }
            else
            {
                return this.mimeType;
            }
        }

        public bool IsValid()
        {
            return this.stream != null && !String.IsNullOrEmpty(this.fileName);
        }

        public void Write(ref Stream output)
        {
            try
            {
                long sLen = stream.Length;
                byte[] buffer = new byte[READ_BUFFER_SIZE];
                int n = 0;
                while (sLen > 0)
                {
                    n = stream.Read(buffer, 0, READ_BUFFER_SIZE);
                    sLen -= n;
                    output.Write(buffer, 0, n);
                }
            }
            finally
            {
                if (stream != null)
                {
                    stream.Close();
                }
            }
        }
    }
}
